app/components/change-phone-number-overlay.js

```diff
@@ -0,0 +1,60 @@
+import Component from "@ember/component";
+import { inject as service } from "@ember/service";
+import { computed } from "@ember/object";
+import AsyncMixin from "browse/mixins/async_tasks";
+
+export default Component.extend(AsyncMixin, {
+  accountService: service(),
+  store: service(),
+  messageBox: service(),
+
+  mobileNumber: "",
+
+  formattedMobileNumber: computed("mobileNumber", function() {
+    const mobile = this.get("mobileNumber");
+    return `+852${mobile}`;
+  }),
+
+  actions: {
+    cancelSearch() {
+      this.get("accountService").setOverlayVisibility(false);
+      this.set("mobileNumber", "");
+      this.set("pin", "");
+      this.set("showPinInput", false);
+    },
+
+    async initiateChange() {
+      try {
+        const data = await this.runTask(
+          this.get("accountService").initiatePinFor(
+            this.get("formattedMobileNumber"),
+            this.get("session.currentUser.id")
```

Comment:
I feel instead of passing `currentUser.id` we can directly use  `id` in service itself.